packages/generator-common/src/options-parser.ts

```diff
@@ -110,6 +110,45 @@ export function resolveRequiredPath<GeneratorOptionsT>(
     : resolve(arg.toString());
 }
 
+/**
+ * Resolves input string using glob notation.
+ * @internal
+ * @param arg - Path argument as passed by the user.
+ * @param options - Options as passed by the user.
+ * @returns Absolute path as a `string`.
+ */
+export function resolveInputGlob<GeneratorOptionsT>(
+  arg: string,
+  options: GeneratorOptionsT & { config?: string }
+): string[] {
+  const inputPath = options.config
+    ? resolve(dirname(options.config), arg.toString())
+    : resolve(arg.toString());
+  return getInputFilePaths(inputPath.split(sep).join(posix.sep));
+}
+
+/**
+ * Recursively searches through a given input path and returns all file paths as a string array.
+ * @param input - the path to the input directory.
+ * @returns all file paths as a string array.
+ * @internal
+ */
+export function getInputFilePaths(input: string): string[] {
+  if (glob.hasMagic(input)) {
```

Comment:
[q]I assume the `hasMagic` means that it is a directory? Please add a comment.

Reply:
Added a comment. `hasMagic` checks if pattern has special characters.